File: examples/player_movement_2d.py
from pyray import *

def main():
    screen_width = 800
    screen_height = 600

    init_window(screen_width, screen_height, "raylib [python] - 2D Player Movement")

    player_pos = Vector2(screen_width / 2.0, screen_height / 2.0)
    player_speed = 5.0
    player_radius = 20.0

    set_target_fps(60)

    while not window_should_close():
        if is_key_down(KEY_RIGHT):
            player_pos.x += player_speed
        if is_key_down(KEY_LEFT):
            player_pos.x -= player_speed
        if is_key_down(KEY_UP):
            player_pos.y -= player_speed
        if is_key_down(KEY_DOWN):
            player_pos.y += player_speed

        if player_pos.x < player_radius:
            player_pos.x = player_radius
        if player_pos.x > screen_width - player_radius:
            player_pos.x = screen_width - player_radius
        if player_pos.y < player_radius:
            player_pos.y = player_radius
        if player_pos.y > screen_height - player_radius:
            player_pos.y = screen_height - player_radius

        begin_drawing()

        clear_background(RAYWHITE)

        draw_circle_v(player_pos, player_radius, RED)

        draw_text("Move the ball with arrow keys", 10, 10, 20, DARKGRAY)
        draw_fps(10, 40)

        end_drawing()

    close_window()

if __name__ == "__main__":
    main()

File: examples/player_movement_2d.c
#include "raylib.h"

int main(void)
{
    const int screenWidth = 800;
    const int screenHeight = 600;

    InitWindow(screenWidth, screenHeight, "raylib - 2D Player Movement");

    Vector2 playerPos = { screenWidth / 2.0f, screenHeight / 2.0f };
    float playerSpeed = 5.0f;
    float playerRadius = 20.0f;

    SetTargetFPS(60);

    while (!WindowShouldClose())
    {
        if (IsKeyDown(KEY_RIGHT)) playerPos.x += playerSpeed;
        if (IsKeyDown(KEY_LEFT)) playerPos.x -= playerSpeed;
        if (IsKeyDown(KEY_UP)) playerPos.y -= playerSpeed;
        if (IsKeyDown(KEY_DOWN)) playerPos.y += playerSpeed;

        if (playerPos.x < playerRadius) playerPos.x = playerRadius;
        if (playerPos.x > screenWidth - playerRadius) playerPos.x = screenWidth - playerRadius;
        if (playerPos.y < playerRadius) playerPos.y = playerRadius;
        if (playerPos.y > screenHeight - playerRadius) playerPos.y = screenHeight - playerRadius;

        BeginDrawing();

            ClearBackground(RAYWHITE);

            DrawCircleV(playerPos, playerRadius, RED);

            DrawText("Move the ball with arrow keys", 10, 10, 20, DARKGRAY);
            DrawFPS(10, 40);

        EndDrawing();
    }

    CloseWindow();

    return 0;
}
